### DISCUSSION TITLE
Dataview tasks not working

### DISCUSSION THREAD

**Discussion by DannyHatcher-Git** [RafaelGB/obsidian-db-folder#361] [General]
I am new to this so apologies if I have added this in the wrong place. Guide me 😁

The dataview task query works for some pages but not others and I don't know why.

**Comment by RafaelGB** [comment]
Hi @dannyHatcher-Git , could you share your task query?

The plugin read all the source files and if their yaml frontmatters are broken in some way, could not be displayed

**Reply by DannyHatcher-Git** [reply]
I have some task queries in the pages but I am using the task metadata option from the db folder plugin (Tasks - enable/disable file tasks column).

Hopefully if the images work the first shows the results for the task query, but in the second you can see that the tasks in the page are not accurate.

![image](https://user-images.githubusercontent.com/78496825/189613980-aef8d6f8-95c5-4736-aff7-fdadba3f93ce.png)

![image](https://user-images.githubusercontent.com/78496825/189614032-a8f1c4a4-fbde-41f8-9ad0-a4d3a24af13f.png)

**Reply by DannyHatcher-Git** [reply]
I also keep getting this error... I don't understand what is going wrong?
![image](https://user-images.githubusercontent.com/78496825/189614295-db3ffac8-c6fd-4229-a3c7-38d5084a2793.png)

**Reply by RafaelGB** [reply]
about the error msg, you can not leave the value of contains filter empty, so when the filter is loaded again is removed automatically.


If you want to filter empty msgs you have the is empty option



About the random task, could you share the console log? maybe something is wrong with the internal dataview query call. ( there is a filter of completed tasks too on column settings)

macos cmd+opt+I 
linux/windows ctrl+shift+I

**Reply by DannyHatcher-Git** [reply]
Thanks for the quick response! I am planning on doing a video with this plugin today so this helps a lot 😁

I am using false in the "Done" checkbox property. There isn't anything empty.

The starting search.
![image](https://user-images.githubusercontent.com/78496825/189615635-b140ee8a-6631-4d4e-bc50-844ae39b73ce.png)
The tasks that should be showing.
![image](https://user-images.githubusercontent.com/78496825/189615715-5f359b52-bf7d-453f-b2b9-8781982003fc.png)

I am not sure how the console log works but this is what I see 
![image](https://user-images.githubusercontent.com/78496825/189616011-2b3a28b6-2ea9-4d00-8657-9b68a5932ba7.png)

**Reply by RafaelGB** [reply]
hmm looking at your menu I think you have to upload the plugin.

There is a breaking change with 2.5.x 

the internal yaml now has another wrappers, if you have any problem saving changes just create a new ddbb

**Reply by DannyHatcher-Git** [reply]
Yeah, I just updated but the saving of filters issue still happens.
The tasks query is still not accurate, some pages showing not tasks where there are tasks, and some pages showing tasks from other pages.

I couldn't find any filter setting for the tasks column, on hide or show completed which I already use.

**Reply by DannyHatcher-Git** [reply]
Maybe it is because I am on v0.16.2?

**Reply by RafaelGB** [reply]
Its tested with 16.2

My clue is the yaml format
- Open the markdown file
- replace the start of the yaml <%% per %% dbfolder:yaml
- replace the end of the yaml %%> per %%


It was a change of 2.5.0 I swear it can be updated automatically but not in all the cases

**Reply by DannyHatcher-Git** [reply]
Yeah. Mine already has that type of frontmatter.

**Reply by DannyHatcher-Git** [reply]
This might help understand what is going on: https://youtu.be/CgH-y-PCuMc
